src/simpleMask.js

```diff
@@ -0,0 +1,171 @@
+const simpleMask = (containerEl, options = {}) => {
+    const defaults = {
+        format: null,
+        inputSelector: null,
+        maskPlaceholder: 'X',
+        numeric: false,
+        alphanumeric: false,
+        customPattern: null,
+        alphanumericPattern: /^[a-zA-Z0-9\.]*$/,
+        numberPattern: /[0-9\/]+/,
+        onMaskCallback: null,
+        onFailedInputCallback: null,
+    };
+
+    const settings = { ...defaults, ...options };
+
+    const maskMethods = {
+        reverse: 'reverse',
+        forward: 'forward'
+    }
+
+    const availableCharacterPatterns =  [settings.numeric, settings.alphanumeric, !!settings.customPattern]
+    const requiredParams = [containerEl, settings.inputSelector, settings.format]
+
+
+    let inputEl;
+    let output;
+    let builtValue;
+    let formatArrayOfCharacters;
+    let arrayOfSpecialCharactersInFormat;
+
+
+    const init = () => {
+        if (settingsAreValid()) {
+            inputEl = containerEl.querySelector(settings.inputSelector);
+            initPatternData();
+            bindEvents();
+        } else {
+            throw new Error('Missing one or many of the following: containerEl, input selector, numeric, alphanumeric or custom pattern')
+        }
+
+    };
+
+    const settingsAreValid = () => {
+        const selectedCharacterPattern = availableCharacterPatterns.filter(pattern => pattern === true);
+        const requiredParamsArePresent = requiredParams.every(param => param);
+
+        let settingsAreValid = true;
+
+        if(selectedCharacterPattern.length !== 1 || !requiredParamsArePresent) {
+            settingsAreValid = false;
+        }
+
+        return settingsAreValid;
+    }
+
+    const initPatternData = () => {
+        formatArrayOfCharacters = settings.format.split('');
```

Comment:
[question] is `formatArrayOfCharacters` used anywhere outside this function? If not, consider scoping the variable to this function instead of the whole plugin

Reply:
> [question] is `formatArrayOfCharacters` used anywhere outside this function? If not, consider scoping the variable to this function instead of the whole plugin

Nope. Change incoming.